src/useradd.c

```diff
@@ -1997,17 +2002,22 @@ static void lastlog_reset (uid_t uid)
 	memzero (&ll, sizeof (ll));
 
 	fd = open (LASTLOG_FILE, O_RDWR);
-	if (   (-1 == fd)
-	    || (lseek (fd, offset_uid, SEEK_SET) != offset_uid)
+	if (-1 == fd) {
+		fprintf (stderr,
+		         _("%s: failed to open the lastlog file for UID %lu: %s\n"),
+		         Prog, (unsigned long) uid, strerror (errno));
+		SYSLOG ((LOG_WARN, "failed to open the lastlog file for UID %lu", (unsigned long) uid));
+		return;
+	}
+	if (   (lseek (fd, offset_uid, SEEK_SET) != offset_uid)
 	    || (write (fd, &ll, sizeof (ll)) != (ssize_t) sizeof (ll))
-	    || (fsync (fd) != 0)
-	    || (close (fd) != 0)) {
+	    || (fsync (fd) != 0)) {
 		fprintf (stderr,
 		         _("%s: failed to reset the lastlog entry of UID %lu: %s\n"),
 		         Prog, (unsigned long) uid, strerror (errno));
 		SYSLOG ((LOG_WARN, "failed to reset the lastlog entry of UID %lu", (unsigned long) uid));
-		/* continue */
 	}
+	close (fd);
```

Comment:
This looks good, but please still check the close() return value.

Reply:
(And yes I know I merged the other one with that check :)

Reply:
Done

Reply:
By the way, I've also changed the same error on `faillog_reset()`.